crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -40,8 +40,21 @@ def role_constraints
   # Unfortunatelly we need to explicitely look at crowbar-status of the proposal
   # because apply_role from this model ignores errors from superclass's apply_role.
   def commit_and_check_proposal
-    proposal_commit("default", false, false)
+    answer = proposal_commit("default", false, false)
+
+    # check if error message is saved in one of the nodes
+    if answer.first != 200
+      NodeObject.find("state:crowbar_upgrade").each do |node|
+        error = node["crowbar_wall"]["chef-error"] || ""
+        next if error.empty?
+        @logger.error("Chef run ended with an error on the node #{name}: #{error}")
+        node["crowbar_wall"]["chef-error"] = ""
+        node.save
```

Comment:
Yes, i was expecting discussion about this. Though this should work, resetting before the call seems more right, yes.